### PR TITLE
psram: refactor psram initialization code into bouffalo-hal

### PR DESCRIPTION
**Pull request by jakezhu9**
In this pull request, I have reorganized and refactored the PSRAM initialization code. I extracted the key part and integrated them into the bouffalo-hal, while removing any unrelated code. This makes it easier for us to use.

4506a524ccd3a5d1a4d730bd045f0c6d

Note: The code contains some magic values, which also appear in this form in the official SDK. For reference, see: https://github.com/bouffalolab/bouffalo_sdk/blob/master/drivers/soc/bl808/std/src/bl808_psram_uhs.c

### CODE REVIEW

File: bouffalo-hal/src/psram.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+/// Initializes the PSRAM.
+#[inline]
+pub fn init_psram<
+    PSRAM: Deref<Target = RegisterBlock>,
+    GLB: Deref<Target = glb::v2::RegisterBlock>,
+>(
+    psram: &PSRAM,
+    glb: &GLB,
+) {
```
Comment (luojia65) [inline bouffalo-hal/src/psram.rs:28]:
```suggestion
pub fn init_psram(
    psram: &RegisterBlock,
    glb: &glb::v2::RegisterBlock,
) {
```

Will this work as well? (BTW, please run `cargo fmt` before commit)

**Review by luojia65 — APPROVED** [review]
It's brilliant! Thanks for your hard work.

**Review by luojia65 — APPROVED** [review]
Great!

File: bouffalo-hal/src/psram.rs (hunk 1 above)
Comment (jakezhu9) [reply bouffalo-hal/src/psram.rs]: done

**Pull request merged**